locks.go

```diff
@@ -143,3 +148,31 @@ func (l *LockRedis) Unlock(ctx context.Context) error {
 	}
 	return nil
 }
+
+// LockMemcached is a wrapper around github.com/alessandro-c/gomemcached-lock that implements the DistLocker interface.
+type LockMemcached struct {
```

Comment:
This is what Google's Gemini told me about implementing a distributed lock via Memcached:

```
It's important to understand that memcached is not ideal for implementing reliable locks due to its inherent limitations:

No guaranteed data retention: Memcached can evict data at any point due to memory pressure, even if it appears to have space available. This can lead to unexpected lock releases.
Lack of distributed locking features: Memcached doesn't offer functionalities like distributed coordination required for consistent locking across multiple servers.
```

And this is ChatGpt4:

```
Also, be aware that memcached, by its nature (being a cache), is not designed for high durability or persistence, so the use cases for such locks might be limited compared to other solutions like etcd, ZooKeeper, or Redis, which offer more robust distributed locking primitives.
```

I think it's worth mentioning those risks somewhere in this doc-comment.

Reply:
Yeah, I did mentioned that in https://github.com/mennanov/limiters/pull/32

Let me update the document about this implication as well.